File: examples/python-examples/Fusion.dox.py
from meshlib import mrmeshpy as mm

points = mm.loadPoints("Points.ply")

params = mm.PointsToMeshParameters()
params.voxelSize = points.computeBoundingBox().diagonal()*1e-2
params.sigma = max(params.voxelSize,mm.findAvgPointsRadius(points,50))
params.minWeight = 1

mesh = mm.pointsToMeshFusion(points,params)

mm.saveMesh(mesh,"Mesh.ctm")

File: examples/js-examples/Fusion.dox.js
import createMeshLib from '@meshinspector/meshlib';

const ml = await createMeshLib();

using pointCloud = ml.PointsLoad.fromAnySupportedFormat('Points.ply');

using bbox = pointCloud.computeBoundingBox();
using params = new ml.PointsToMeshParameters();
params.voxelSize = bbox.diagonal() * 0.01;
params.sigma = Math.max(params.voxelSize, ml.findAvgPointsRadius(pointCloud, 50));
params.minWeight = 1;

using mesh = ml.pointsToMeshFusion(pointCloud, params);

ml.MeshSave.toAnySupportedFormat(mesh, 'Mesh.ply');
